app/src/main/java/zapsolutions/zap/fragments/WalletFragment.java

```diff
@@ -245,7 +246,13 @@ public void onSingleClick(View v) {
                 mWalletConnectedLayout.setVisibility(View.GONE);
                 mWalletNotConnectedLayout.setVisibility(View.GONE);
                 mLoadingWalletLayout.setVisibility(View.VISIBLE);
+
+                if(TorUtil.isCurrentConnectionTor()){
+                    TorUtil.askToInstallOrbotIfMissing(getActivity());
+                }
+
                 Wallet.getInstance().checkIfLndIsReachableAndTriggerWalletLoadedInterface();
```

Comment:
If its a tor-connection and Orbot is not installed, do we still want to execute this method?
Maybe we could remove the `ifMissing` part from the method above and do something like:
```
if(TorUtil.isCurrentConnectionTor() && !TorUtil.isInstalled(getActivity())){
   TorUtil.askToInstallOrbot(getActivity());
} else {
   // no tor, or orbot is installed
   Wallet.getInstance().checkIfLndIsReachableAndTriggerWalletLoadedInterface();
}
```